proot-distro.sh

```diff
@@ -814,6 +814,54 @@ command_remove_help() {
 	show_version
 	msg
 }
+command_clear_cache_help() {
+	msg
+	msg "${BYELLOW}Usage: ${BCYAN}$PROGRAM_NAME ${GREEN}clear-cache"
+	msg
+	msg "${CYAN}This command will delete the download cache...${RST}"
+	msg
+	msg "${CYAN}This command is useful to clear some space by deleting dowmloaded rootfs tarballs${RST}"
+	msg
+	show_version
+	msg
+}
+
+#############################################################################
+#
+# FUNCTION TO CLEAR DLCACHE
+#
+
+command_clear_cache() {
+	if [ $# -ge 1 ]; then
+		case "$1" in
+			-h|--help) command_clear_cache_help;;
+			*)
+				msg
+				msg "${BRED}Error: unknown command '${YELLOW}$1${BRED}'.${RST}"
+				msg
+				msg "${CYAN}Run '${GREEN}${PROGRAM_NAME} clear-cache --help${CYAN}' to see the list of available commands.${RST}"
+				msg
+				exit 1
+			;;
+		esac
+	elif (( $# == 0 )); then
+
+		files="$(find "${DOWNLOAD_CACHE_DIR}" -maxdepth 1 -type f \( -iname "*.tar.gz" -o -iname "*.taz" -o -iname "*.tgz" -o -iname "*.tar.bz2" -o -iname "*.tb2" -o -iname "*.tbz" -o -iname "*.tbz2" -o -iname "*.tz2" -o -iname "*.tar.lz" -o -iname "*.tar.lzma" -o -iname "*.tlz" -o -iname "*.tar.lzo" -o -iname "*.tar.xz" -o -iname "*.txz" -o -iname "*.tar.Z" -o -iname "*.tz" -o -iname "*.taz" -o -iname "*tar.zst" -o -iname "*.tar.zst" -o -iname "*.lz" -o -iname "*.lzo" \))"
+
+		if [ -z "${files}" ]; then
+			msg "${BLUE}[${GREEN}*${BLUE}] ${CYAN}No Download cache found...${RST}"
+		else
+			SIZE_OF_CACHE="$(du -d 0 -h -a ${DOWNLOAD_CACHE_DIR} | awk '{$2=$2};1' | cut -d " " -f 1)"
+			msg "${BLUE}[${GREEN}*${BLUE}] ${CYAN}Clearing cache files...${RST}"
+			
+			for f in ${files}; do
+				msg "${BLUE}[${GREEN}*${BLUE}] ${CYAN} Removing ${BLUE} ${f} ${RST}"
+				rm -rf "${f}"
+			done
+			msg "${BLUE}[${GREEN}*${BLUE}] ${CYAN} \"${SIZE_OF_CACHE}\" Of cache removed ${RST}"
```

Comment:
You're cleaning download cache, isn't it? So why you are selecting files by its extension? Latest revision has ultra-long file command specifying all (or not?) possible extensions.

Use while loop + `find`. What I mean exactly:
```
while read -r filename; do
    msg "${BLUE}[${GREEN}*${BLUE}] ${CYAN} Removing ${BLUE} ${filename} ${RST}"
    rm -f "${filename}"
done < <(find "${DOWNLOAD_CACHE_DIR}" -type f)
```

Reply:
**Reason why I made that extremly large find**  :  when I studied that script its just storing fs files into it. And I thought it's not proper thing to delete everything from that directory in that case cause if user keeps something in cache folder (intentionally or by mistake) it gets deleted to by printing out  `hey I deleted this file too` it looks bad. 
If dl_cache directory is strictly for cache then using while loop will be good as you said ..

Reply:
- Switching for while..